File: trunk/OscarDotNet/component/biz/Class_biz_sections.pas
unit Class_biz_sections;

interface

type
  filter_type = 0..14;
  TClass_biz_sections = class
  private
    { Private Declarations }
  public
    constructor Create;
    procedure BindListControl
      (
      target: system.object;
      selected_description: string = ''
      );
  end;

implementation

uses
  mysql.data.mysqlclient,
  system.web.ui.webcontrols;

constructor TClass_biz_sections.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

procedure TClass_biz_sections.BindListControl
  (
  target: system.object;
  selected_description: string = ''
  );
var
  i: cardinal;
begin
  DropDownList(target).items.Clear;
  if selected_description = EMPTY then begin
    DropDownList(target).Items.Add(listitem.Create('-- Select --',''));
  end;
  DropDownList(target).Items.Add(listitem.Create('*','0'));
  for i := 1 to 14 do begin
    DropDownList(target).Items.Add(listitem.Create(i.tostring,i.ToString));
    if i.tostring = selected_description then begin
      DropDownList(target).selectedvalue := i.tostring;
    end;
  end;
end;

end.

File: trunk/OscarDotNet/component/biz/Class_biz_sections.cs
using kix;
using System.Configuration;
using System.Web.UI.WebControls;

namespace Class_biz_sections
  {
  public class TClass_biz_sections
    {

    public TClass_biz_sections() : base()
      {
      }

    public void BindListControl
      (
      object target,
      string selected_description = k.EMPTY
      )
      {
      ((target) as DropDownList).Items.Clear();
      if (selected_description.Length == 0)
        {
        ((target) as DropDownList).Items.Add(new ListItem("-- Select --", ""));
        }
      ((target) as DropDownList).Items.Add(new ListItem("*", "0"));
      for (var i = new k.subtype<int>(1,int.Parse(ConfigurationManager.AppSettings["max_section_num"]) + 1); i.val < i.LAST; i.val++)
        {
        ((target) as DropDownList).Items.Add(new ListItem(i.val.ToString(), i.val.ToString()));
        if (i.val.ToString() == selected_description)
          {
          ((target) as DropDownList).SelectedValue = i.val.ToString();
          }
        }
      }

    } // end TClass_biz_sections

  }
